x-pack/plugins/reporting/server/browsers/chromium/driver_factory/index.ts

```diff
@@ -66,18 +67,7 @@ export class HeadlessChromiumDriverFactory {
       proxy: this.browserConfig.proxy,
     });
 
-    return puppeteerLaunch({
-      userDataDir: this.userDataDir,
-      executablePath: this.binaryPath,
-      ignoreHTTPSErrors: true,
-      args: chromiumArgs,
-    } as LaunchOptions).catch((error: Error) => {
-      logger.error(
-        `The Reporting plugin encountered issues launching Chromium in a self-test. You may have trouble generating reports.`
-      );
-      logger.error(error);
-      return null;
-    });
+    return browserTest(this.binaryPath, chromiumArgs, logger);
```

Comment:
Maybe have the whole body of this function in the `routes/diagnose/browser` handler? I'm not sure this `test` method needs to be on this class. The scope here has class members for the binary path and capture config, but I think we can get those in the route handler as well.